Lib/platform.py

```diff
@@ -334,6 +334,28 @@ def _syscmd_ver(system='', release='', version='',
     (6, None): "post2012ServerR2",
 }
 
+def win32_is_iot():
+    return win32_edition() in ('IoTUAP', 'NanoServer', 'WindowsCoreHeadless', 'IoTEdgeOS')
+
+def win32_edition():
+    try:
+        from winreg import OpenKeyEx, QueryValueEx, CloseKey, HKEY_LOCAL_MACHINE
+    except ImportError:
+        from _winreg import OpenKeyEx, QueryValueEx, CloseKey, HKEY_LOCAL_MACHINE
+
+    key = None
+    try:
+        key = OpenKeyEx(HKEY_LOCAL_MACHINE,
+                        r'SOFTWARE\Microsoft\Windows NT\CurrentVersion')
+        return QueryValueEx(key, 'EditionId')[0]
```

Comment:
Use (properly formatted) instead of the try/finally (you'll still need the try/except):

```python
with OpenKeyEx(HKEY_LOCAL_MACHINE, r'SOFTWARE\Microsoft\Windows NT\CurrentVersion') as key:
    return QueryValueEx(key, 'EditionId')[0]
```

Reply:
I just fixed this in #12912, which will fix your Ubuntu and macOS tests that broke.

Reply:
Should I rebase and force-push to pick up #12912?